src/Common/dataAccess/readCollection.ts

```diff
@@ -22,3 +26,51 @@ export async function readCollection(databaseId: string, collectionId: string):
   clearMessage();
   return collection;
 }
+
+export async function getMongoCollectionIndexTransformationProgress(
+  databaseId: string,
+  collectionId: string
+): Promise<number> {
+  let indexTransformationPercentage: number;
+  const clearMessage = logConsoleProgress(`Reading container ${collectionId}`);
+  try {
+    const response = await client()
+      .database(databaseId)
+      .container(collectionId)
+      .read({ populateQuotaInfo: true });
+
+    indexTransformationPercentage = parseInt(
+      response.headers[Constants.HttpHeaders.collectionIndexTransformationProgress] as string
+    );
+  } catch (error) {
+    logConsoleError(`Error while reading container ${collectionId}:\n ${JSON.stringify(error)}`);
```

Comment:
JSON.stringify(error) is never what you want. It will log `{}` since errors have no enumerable properties. @victor-meng is fixing this up in other places and I will try to add a linting rule

Reply:
will sync with him to make those changes.